tests/tools/test_edges_with_features.py

```diff
@@ -0,0 +1,320 @@
+
+            # get orig_eids
+            orig_type_eids = g.edata['orig_id'].numpy()
+            orig_etype_ids = g.edata[dgl.ETYPE].numpy()
+            print(f'[Rank: {i}] orig_etype_ids: {np.bincount(orig_etype_ids)}')
+
+
+        for i in range(num_chunks):
+            sub_dir = 'part-' + str(i)
+            assert meta_data[sub_dir]['node_feats'] == 'part{}/node_feat.dgl'.format(i)
+            assert meta_data[sub_dir]['edge_feats'] == 'part{}/edge_feat.dgl'.format(i)
+            assert meta_data[sub_dir]['part_graph'] == 'part{}/graph.dgl'.format(i)
+
+            # check data
+            sub_dir = os.path.join(out_dir, 'part' + str(i))
+
+            # Read graph.dgl
+            fname = os.path.join(sub_dir, 'graph.dgl')
+            assert os.path.isfile(fname)
+            g_list, data_dict = load_graphs(fname)
+            g = g_list[0]
+            assert isinstance(g, dgl.DGLGraph)
+
+            # Read node_feat.dgl
+            fname = os.path.join(sub_dir, 'node_feat.dgl')
+            assert os.path.isfile(fname)
+            tensor_dict = load_tensors(fname)
+            all_tensors = ['paper/feat', 'paper/label', 'paper/year']
+            assert tensor_dict.keys() == set(all_tensors)
+            for key in all_tensors:
+                assert isinstance(tensor_dict[key], torch.Tensor)
+
+            # Read edge_feat.dgl
+            fname = os.path.join(sub_dir, 'edge_feat.dgl')
+            assert os.path.isfile(fname)
+            tensor_dict = load_tensors(fname)
+            all_tensors = ['paper:cites:paper/count', 
+                           'author:writes:paper/year', 
+                           'paper:rev_writes:author/year']
+            assert tensor_dict.keys() == set(all_tensors)
+            for key in all_tensors:
+                assert isinstance(tensor_dict[key], torch.Tensor)
+
+            # Get orig_eids
+            orig_type_eids = g.edata['orig_id'].numpy()
+            orig_etype_ids = g.edata[dgl.ETYPE].numpy()
+
+            # Compare the data stored as edge features in this partition with the data
+            # from the original graph.
+            etype_names = list(edge_dict.keys())
+            for idx, etype_name in enumerate(etype_names): 
+                part_data = None
+                key = None
+                if etype_name+'/count' in tensor_dict: 
+                    key = etype_name+'/count' 
+                    part_data = tensor_dict[etype_name+'/count'].numpy()
+                if etype_name+'/year' in tensor_dict: 
+                    key = etype_name+'/year' 
+                    part_data = tensor_dict[etype_name+'/year'].numpy()
+
+                if part_data is None: 
+                    continue
+
+                gold_type_ids = orig_type_eids[ orig_etype_ids == idx ]
+                gold_data = edge_data_gold[key][gold_type_ids]
+                assert np.all(gold_data == part_data)
+
+
+if __name__ == '__main__':
+    test_edges_with_features()
```

Comment:
I feel a lot of the code is duplicate of https://github.com/dmlc/dgl/blob/master/tests/tools/test_dist_part.py . You can add a new test case in the same file and try to reuse the test graph construction code.

Reply:
Yes. The dataset is almost same as in the test_dist_part.py, but there are a couple of changes which are different compared to the case described in test_dist_part.py. I still haven't figured out how to wrap this common code in a function so that it can be used by most of the test cases. 

I propose the following. We can solve this issue as a separate issue during code refactoring. Also, we still haven't figured out how to run same test case in different configurations like changing some parameters like no. of nodes/edges/features-types etc... 

Given this type of test cases, it will be hard to put everything in a function and re-use this function for some of the test cases which would need this. 

For now, please approve this PR., and I will discuss this suggestion with Rui Ying and figure out what is the best possible way to refactor this code.

Reply:
For tests, it is acceptable that we lower our bar for early checkin, also it is not harmful to copy-paste code in tests.

Reply:
Frankly speaking, so much duplicate code is not acceptable for me, even in test. And I don't see any blocks to add check for edge feature into existing `test_dist_part.py`. 

Here's what I am thinking: 

In this PR, pls add tests into existing one though the existing one is far from complete from my perspective.
 
as for the more comprehensive/elegant test for dist part pipeline, I recommend to unify it with `//tests/distributed/test_partition.py` which is for `dgl.distributed.partition_graph()`. 

I have already tried with it and found much part could be shared, especially the verification of partitioned graphs. we should test the partitions could be successfully loaded via `dgl.distributed.load_partition()`. This should be the key part.

Reply:
Done. I have moved the code which was replicated and created a new file/module for this function. Moving forward whoever wants to use this code can call this function directly.

Reply:
Great feedback, thanks @Rhett-Ying  for raising the bar!